src/auth/bearer/Token.h

```diff
@@ -0,0 +1,45 @@
+#ifndef _SQUID_SRC_AUTH_BEARER_TOKEN_H
+#define _SQUID_SRC_AUTH_BEARER_TOKEN_H
+
+#if HAVE_AUTH_MODULE_BEARER
+
+#include "auth/bearer/forward.h"
+#include "base/RefCount.h"
+#include "sbuf/SBuf.h"
+#include "SquidTime.h"
+
+namespace Auth {
+namespace Bearer {
+
+/// a Bearer token we have seen and details we associate with it
+class Token : public RefCountable
+{
+    MEMPROXY_CLASS(Auth::Bearer::Token);
+
+public:
+    Token() : expires(squid_curtime) {}
+    explicit Token(const SBuf &token) : b68encoded(token), expires(squid_curtime) {}
+    ~Token();
+
+    /// the B68 encoding form of this token
+    SBuf b68encoded;
+
+    /// the Auth::User this token is tied to
+    Auth::Bearer::User *user = nullptr;
+
+    /// when this token will expire
+    time_t expires = 0;
+
+    /// a cache of known tokens
+    static TokenCache Cache;
+
+private:
+    Token(const Token &); // do not implement
```

Comment:
Yes I am an advocate of that practice, however past audits have had significant resistance about that practice. So the code as-is was changed to meet those auditor requests.